lib/services/preference.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2017 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+)
+
+// ClusterAuthPreference defines an interface to get and set
+// authentication preferences for a cluster.
+type ClusterAuthPreference interface {
+	GetClusterAuthPreference() (AuthPreference, error)
+	SetClusterAuthPreference(AuthPreference) error
+}
+
+// AuthPreference defines the authentication preferences for a specific
+// cluster. It defines the type (local, oidc) and second factor (off, otp, oidc).
+type AuthPreference interface {
+	GetType() string
+	SetType(string)
+	GetSecondFactor() string
+	SetSecondFactor(string)
+}
```

Comment:
Add a `Check() error` method here, then in `config/fileconfig.go` during `Parse()` you can call `Check()` to make sure the configuration is valid.